lib/rpush/client/active_model/fcm/notification.rb

```diff
@@ -0,0 +1,121 @@
+module Rpush
+  module Client
+    module ActiveModel
+      module Fcm
+        module Notification
+          FCM_PRIORITY_HIGH = Rpush::Client::ActiveModel::Apns::Notification::APNS_PRIORITY_IMMEDIATE
+          FCM_PRIORITY_NORMAL = Rpush::Client::ActiveModel::Apns::Notification::APNS_PRIORITY_CONSERVE_POWER
+          FCM_PRIORITIES = [FCM_PRIORITY_HIGH, FCM_PRIORITY_NORMAL]
+
+          ROOT_NOTIFICATION_KEYS = %w[title body image].freeze
+          ANDROID_NOTIFICATION_KEYS = %w[icon tag color click_action body_loc_key body_loc_args title_loc_key
+                                         title_loc_args channel_id ticker sticky event_time local_only
+                                         default_vibrate_timings default_light_settings vibrate_timings
+                                         visibility notification_count light_settings sound].freeze
+
+          def self.included(base)
+            base.instance_eval do
+              validates :device_token, presence: true
+              validates :priority, inclusion: { in: FCM_PRIORITIES }, allow_nil: true
+
+              validates_with Rpush::Client::ActiveModel::PayloadDataSizeValidator, limit: 4096
+              validates_with Rpush::Client::ActiveModel::RegistrationIdsCountValidator, limit: 1000
+
+              validates_with Rpush::Client::ActiveModel::Fcm::ExpiryCollapseKeyMutualInclusionValidator
+              validates_with Rpush::Client::ActiveModel::Fcm::NotificationKeysInAllowedListValidator
```

Comment:
[Question]: This is kinda hard to track.. I believe this validator ensures that only keys defined in `ROOT_NOTIFICATION_KEYS` and `ANDROID_NOTIFICATION_KEYS` are allowed, which GCM _didn't_ protect against. I think this lines up with: https://firebase.google.com/docs/reference/fcm/rest/v1/projects.messages#androidnotification

That being said, how do we set `data`. I don't see a setter, but `data` is where our "custom attributes" would go right? According to: https://firebase.google.com/docs/reference/fcm/rest/v1/projects.messages#AndroidNotification

For example, you set `notification.data` here, but does this work?
https://github.com/Codefied/housecall-web/pull/51049/files#diff-9c910a3c8aebb4bc9d2dd56c81d61af7a4eb808083857215c37c39f87e268db7R192-R194

Reply:
what work is `notification.notification = data`, still testing, need to exclude the our custom attr (outside of AndroidNotification) from `notification.notification`

Reply:
I see where data gets set. It's in the active record base class: https://github.com/Codefied/rpush/blob/JS-7167-add-fcm/lib/rpush/client/active_record/notification.rb/#L4 

So both are settable from this object.